system-x/services/keycloak/src/main/java/software/tnb/keycloak/resource/openshift/OpenshiftKeycloak.java

```diff
@@ -0,0 +1,397 @@
+package software.tnb.keycloak.resource.openshift;
+
+import software.tnb.common.deployment.OpenshiftDeployable;
+import software.tnb.common.deployment.WithExternalHostname;
+import software.tnb.common.deployment.WithInClusterHostname;
+import software.tnb.common.deployment.WithName;
+import software.tnb.common.deployment.WithOperatorHub;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.utils.WaitUtils;
+import software.tnb.keycloak.resource.openshift.generated.Keycloak;
+import software.tnb.keycloak.resource.openshift.generated.KeycloakSpec;
+import software.tnb.keycloak.resource.openshift.generated.KeycloakStatus;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.Db;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.Hostname;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.Http;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.Unsupported;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.db.PasswordSecret;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.db.UsernameSecret;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.unsupported.PodTemplate;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.unsupported.podtemplate.Spec;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.unsupported.podtemplate.spec.Containers;
+import software.tnb.keycloak.resource.openshift.generated.keycloakspec.unsupported.podtemplate.spec.containers.Env;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
```

Comment:
usually we remove the `@org.jetbrains.annotations.NotNull` annotations, it is up to you